core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1447,7 +1499,56 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, time: Time, met
     }
   }
 
-  case class TopicDeletionStopReplicaResult(stopReplicaResponseObj: AbstractResponse, replicaId: Int) extends ControllerEvent {
+  case class LeaderAndIsrResponseReceived(LeaderAndIsrResponseObj: AbstractResponse, brokerId: Int) extends ControllerEvent {
```

Comment:
Great point. I have moved the overlapping logic to a new method named `onReplicaBecomeOffline(newOfflineReplicas: Set[PartitionAndReplica])`.